src/github.ts

```diff
@@ -281,6 +284,67 @@ export class GitHub {
     }
   }
 
+  async commitsSinceShaRest(
+    sha?: string,
+    path?: string,
+    per_page = 100
+  ): Promise<Commit[]> {
+    let page = 1;
+    let found = false;
+    const baseBranch = await this.getDefaultBranch();
+    const commits = [];
+    while (!found) {
+      const response = await this.request(
+        'GET /repos/{owner}/{repo}/commits{?sha,page,per_page,path}',
+        {
+          owner: this.owner,
+          repo: this.repo,
+          sha: baseBranch,
+          page,
+          per_page,
+          path,
+        }
+      );
+      for (const commit of (response as CommitsListResponse).data) {
+        if (commit.sha === sha) {
+          found = true;
+          break;
+        }
+        // skip merge commits
+        if (commit.parents.length === 2) {
+          continue;
+        }
+        commits.push([commit.sha, commit.commit.message]);
+      }
+      page++;
+    }
+    const ret = [];
+    for (const [ref, message] of commits) {
+      const files = [];
+      let page = 1;
+      let moreFiles = true;
+      while (moreFiles) {
+        const response = (await this.request(
+          'GET /repos/{owner}/{repo}/commits/{ref}{?page}',
+          {owner: this.owner, repo: this.repo, ref, page}
+        )) as CommitGetResponse;
+        const commitFiles = response.data.files;
+        if (!commitFiles) {
+          moreFiles = false;
+          break;
+        }
+        files.push(...commitFiles.map(f => f.filename ?? ''));
+        if (commitFiles.length < 300) {
```

Comment:
I found this slightly confusing, does the API contract say that it will return up to 300 files in the first request?

Mind just adding a comment that links to the API docs describing the limits of this API? I think there's also an upper-bound limit of 3000 files (unlikely for many people to bump into).

Reply:
done